verifiers/plonky2/src/lib.rs

```diff
@@ -0,0 +1,84 @@
+#![cfg_attr(not(feature = "std"), no_std)]
+extern crate alloc;
+
+use alloc::vec::Vec;
+use core::marker::PhantomData;
+use frame_support::traits::ConstU32;
+use frame_support::weights::Weight;
+use frame_support::BoundedVec;
+use hp_verifiers::{Cow, Verifier, VerifyError};
+use plonky2_verifier::validate::validate_vk_default_poseidon;
+use plonky2_verifier::verify_default_poseidon;
+
+pub mod benchmarking;
+mod resources;
+pub(crate) mod verifier_should;
+mod weight;
+
+pub use weight::WeightInfo;
+
+pub const MAX_VK_SIZE: u32 = 262_144; // 256 KB in bytes
+
+pub type Pubs = Vec<u8>;
+pub type Proof = Vec<u8>;
```

Comment:
Why not enforcing a maximum size also on Pubs and Proofs ?

Reply:
Primarily because `Vk` must be limited, since it is constrained by traits we need it to implement. That is the reason we use `BoundedVec` instead of `Vec` as in other types. I assume the idea is to allow a bigger range of proofs users can verify. Maybe you know whether 256 KB is sufficient?

Reply:
It would be wise to put a limit on proof and pubs as well. I think 256kB might be fine, perhaps we can expose this as configuration parameter ?